core/include/gnuradio-4.0/Block.hpp

```diff
@@ -2024,6 +2037,8 @@ std::string reflFirstTypeName() {
         return fmt::format("gr::DataSet<{}>", reflFirstTypeName<typename Type::value_type>());
     } else if constexpr (UncertainValueLike<Type>) {
         return fmt::format("gr::UncertainValue<{}>", reflFirstTypeName<typename Type::value_type>());
+    } else if constexpr (is_complex_v<Type>) {
+        return fmt::format("std::complex<{}>", reflFirstTypeName<typename Type::value_type>());
```

Comment:
You can also use `pmtv::Complex` concept here:
`if constexpr (pmtv::Complex<Type>)`

Reply:
Looking at https://github.com/gnuradio/pmt/blob/main/include/pmtv/type_helpers.hpp#L67, that concept seems insufficient if someone attempts to use something like `std::complex<int16_t>`, which can happen when doing fixed-point arithmetic implementations. But in any case I'll do this change, since I didn't know that `pmtv` had a `Complex` concept and it doesn't make sense to duplicate it here. If needed, the `pmtv::Complex` concept can be extended.

Reply:
I believe `std::complex` only supports value types of `float`, `double`, and `long double`.

Reply:
It also supports integer types, and it works as you would expect: https://godbolt.org/z/soxf5c6oG

Reply:
Updated to use `pmtv::Complex` and force-pushed.

Reply:
Even if it is possible to instantiate `std::complex` with integer types like `int16_t`, doing so is generally discouraged because complex number operations, especially those involving trigonometric or transcendental functions, rely on floating-point precision. Using integer types can lead to truncation, overflow, and incorrect results for functions like `std::sin()`, which are not designed to handle complex integers. The provided code, while it compiles, may exhibit undefined behavior or errors when performing certain operations due to the lack of proper floating-point precision. 

In any case, this can be discussed if one wants to include complex integer support in the `pmt` library, but it is probably out-of-scope for this PR.